ros2_lifecycle_manager/src/ros2_lifecycle_manager.cpp

```diff
@@ -99,38 +111,132 @@ namespace ros2_lifecycle_manager
+
+
+  uint8_t Ros2LifecycleManager::transition_node_to_state(const uint8_t state, const std::string& node, const std_nanosec &connection_timeout, const std_nanosec &call_timeout)
   {
-    return transition_multiplex(lifecycle_msgs::msg::Transition::TRANSITION_CONFIGURE, ordered, connection_timeout, call_timeout);
+    auto it = node_map_.find(node);
+    if (it == node_map_.end()) // Check if the requested node is being managed
+    {
+
+      RCLCPP_ERROR_STREAM(
+          node_logging_->get_logger(), "State for node: " << node << " could not be provided as that node was not being managed. ");
+
+      return lifecycle_msgs::msg::State::PRIMARY_STATE_UNKNOWN;
+    }
+
+    static constexpr auto UNKNOWN = lifecycle_msgs::msg::State::PRIMARY_STATE_UNKNOWN;
+    static constexpr auto UNCONFIGURED = lifecycle_msgs::msg::State::PRIMARY_STATE_UNCONFIGURED;
+    static constexpr auto INACTIVE = lifecycle_msgs::msg::State::PRIMARY_STATE_INACTIVE;
+    static constexpr auto ACTIVE = lifecycle_msgs::msg::State::PRIMARY_STATE_ACTIVE;
+    static constexpr auto FINALIZED = lifecycle_msgs::msg::State::PRIMARY_STATE_FINALIZED;
+
+    if (state != UNCONFIGURED
+      && state != INACTIVE
+      && state != ACTIVE
+      && state != FINALIZED) // Check if target state is no primary state
+    {
+      RCLCPP_ERROR_STREAM(
+          node_logging_->get_logger(), "transition_node_to_state does not support transitioning to temporary states.");
+      
+      return get_managed_node_state(node); // Return whatever the current state is
+    }
+
+    ///// Setup transition paths /////
+    using TransitionFunc = std::function<std::vector<std::string>(const std_nanosec &, const std_nanosec &, bool, std::vector<std::string>)>;
+
+    namespace stdp = std::placeholders;
+
+    static const auto configure_func = std::bind(&Ros2LifecycleManager::configure, this, stdp::_1, stdp::_2, stdp::_3, stdp::_4);
+    static const auto activate_func = std::bind(&Ros2LifecycleManager::activate, this, stdp::_1, stdp::_2, stdp::_3, stdp::_4);
+    static const auto deactivate_func = std::bind(&Ros2LifecycleManager::deactivate, this, stdp::_1, stdp::_2, stdp::_3, stdp::_4);
+    static const auto cleanup_func = std::bind(&Ros2LifecycleManager::cleanup, this, stdp::_1, stdp::_2, stdp::_3, stdp::_4);
+    static const auto shutdown_func = std::bind(&Ros2LifecycleManager::shutdown, this, stdp::_1, stdp::_2, stdp::_3, stdp::_4);
+
+    // This object represents a mapping of all source->target combinations for primary states to the function call path required to reach that state
+    // It is implemented as a map of maps where the keys for each map are source state and target state respectively
+    // The inner map contains a list of transition functions to call in order to move a node to the target state from the source state 
+    static std::unordered_map<uint8_t, std::unordered_map<uint8_t, std::vector<TransitionFunc>>> transitions_paths = {
```

Comment:
Not sure how easy it is to implement given the static elements you're working with, but this might be easier to read/modify if you define a local struct containing {INITIAL_STATE, TARGET_STATE, TRANSITION_FUNC}. Or use std::tuple, either or. Then you could simply have this map implemented as

```
  std::unordered_map<std::tuple<uint8_t, uint8_t, std::vector<TransitionFunc>>> transition_paths = {
    { UNCONFIGURED, INACTIVE, configure_func },
    { UNCONFIGURED, ACTIVE, {configure_func, activate_func}},
    ...
  }
```
I might even recommend swapping the vector of functions with a simple lambda calling one, then the other, or something, so that it's always a single "function" transitioning from one state to the other, rather than some having multiple.

I don't have a real problem with this whole setup though, I just think it could be a bit more straightforward.